vision_agent/tools/tools.py

```diff
@@ -1878,6 +1878,74 @@ def closest_box_distance(
     return cast(float, np.sqrt(horizontal_distance**2 + vertical_distance**2))
 
 
+def document_analysis(image: np.ndarray) -> Dict[str, Any]:
+    """'document_analysis' is a document understanding tool that can handle and analyse various
+    types of document images with text, tables, pictures, charts and other information.
+    It returns a structured output containing the text, tables, pictures, charts and information
+    caption, summary, labels, bounding boxes avoiding information loss.
+
+    Parameters:
+        image (np.ndarray): The document image to analyze
+
+    Returns:
+        Dict[str, Any]: A dictionary containing the extracted text, tables, and other
+            information.
+
+    Example
+    -------
+        >>> document_analysis(image)
+        {'filename': None,
+         'pages': [{'bbox': [left, top, right, bottom],
+                    'chunks': [{'bbox': [left, top, right, bottom],
+                                'caption': 'TITLE',
+                                'label': 'page_header',
+                                'summary': 'The image contains a single word:\n'
+                                        '\n'
+                                        'TITLE\n'
+                                        '\n'
+                                        'The logo or brand name, '
+                                        'presented in a simple, bold, all-capital '
+                                        'letter font against a plain background.'},
+                                {'bbox': [left, top, right, bottom],
+                                 'caption': {'annotation': 'Cumulative Cloud CapEx Spending',
+                                    'data': [{'value': 200, 'year': '2024'},
+                                             {'value': 250, 'year': '2025'}],
+                                    'notes': 'Values for 2024 and 2025 are '
+                                             "estimates, indicated by 'E'",
+                                    'title': 'Total CapEx Spending',
+                                    'trend': 'Generally increasing over time',
+                                    'type': 'bar chart',
+                                    'unit': 'Billion USD',
+                                    'xAxis': 'Year',
+                                    'yAxis': 'Total CapEx Spending'},
+                                'label': 'picture',
+                        '       summary': 'This bar chart illustrates the trend of '
+                                   'Total CapEx (Capital Expenditure) Spending '
+                                   'over time, from 2024 to 2025.'},
+
+                    (...)
+
```

Comment:
I would make this a shorter, so try to keep the entire doc similar size to other tools. The longer the doc, the more likely the LLM is to choose it.

Reply:
Made it shorter keeping the chart example, I can also remove it if it still too long